### PR TITLE
adds support for renaming a table

### PR DESCRIPTION
**Pull request by seut**
renaming of a partitioned table is not yet supported and will throw an exception. will be done in a separate commit/PR.

### CODE REVIEW

File: sql/src/main/java/io/crate/executor/transport/ddl/RenameTableRequest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public TableIdent targetIdent() {
+        return targetIdent;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (sourceIdent == null && targetIdent == null) {
```
Comment (andreidan) [inline sql/src/main/java/io/crate/executor/transport/ddl/RenameTableRequest.java]: This should be an `or` ( || ) operation right?

File: sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                                      ThreadPool threadPool,
+                                      ActionFilters actionFilters,
+                                      IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(settings, ACTION_NAME, transportService, clusterService, threadPool, actionFilters,
+            indexNameExpressionResolver, RenameTableRequest::new);
+    }
+
+    @Override
+    protected String executor() {
+        // no need to use a thread pool, we go async right away
```
Comment (andreidan) [inline sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java]: This comment is confusing. If we don't use a thread pool we go *sync* right? Or is it meant to say we're *already* async?

File: sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    protected RenameTableResponse newResponse() {
+        return new RenameTableResponse();
+    }
+
+    @Override
+    protected void masterOperation(RenameTableRequest request,
+                                   ClusterState state,
+                                   ActionListener<RenameTableResponse> listener) throws Exception {
+        // validate the target table name
+        // done here, so we can tests that the source table will be re-opened also after rename failures
```
Comment (andreidan) [inline sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java]: can you please move "done here" to the previous line as it's confusing to read it as separate comments. Also there's a small type "so we can tests" -> "so we can test"

File: sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java [hunk 4]
```diff
@@ -191,6 +151,47 @@ public void fail(@Nonnull Throwable t) {
...
 
+    public CompletableFuture<Long> executeAlterTableRenameTable(AlterTableRenameAnalyzedStatement statement) {
+        DocTableInfo sourceTableInfo = statement.sourceTableInfo();
+        TableIdent sourceTableIdent = sourceTableInfo.ident();
+        TableIdent targetTableIdent = statement.targetTableIdent();
+
+        return closeTable(sourceTableInfo)
+            .thenCompose(r -> renameTable(sourceTableIdent, targetTableIdent))
+                .whenComplete((r, f) -> {
+                    TableIdent tableToOpen = targetTableIdent;
```
Comment (Autophagy) [inline sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java]: You should track the state of a table here, and only re-open the table if the state was already open to begin with. If you are renaming a table that is _already_ closed, this operation would re-open the table, which is undesirable.

File: sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java (hunk 2 above)
Comment (seut) [inline sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java]: no the operation will go async as the cluster state task issued at this operation is async

File: sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java (hunk 4 above)
Comment (seut) [inline sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java]: hm ok, currently there is no way to close a table.. but will do

Comment (seut) [inline sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java]: I'll add a FIXME as the related information if a table is closed will be introduced by another PR

**Pull request merged**